### PR TITLE
Added support for `eth_` on `akula-rpc`

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

File: src/rpc/mod.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            types::BlockId::Number(n) => {
+                let block_number =
+                    get_block_number(txn, ethereum_jsonrpc::types::BlockId::Number(n))?;
+                let block_hash = txn.get(tables::CanonicalHeader, block_number)?.unwrap();
+
+                match uncle_index {
+                    Some(n) => {
+                        let body = txn
+                            .get(tables::BlockBody, (block_number, block_hash))?
+                            .unwrap();
```
Comment (vorot93) [inline src/rpc/mod.rs:51]: Please check unwrap usage. Invalid RPC parameters should be handled properly, not bump into unwraps.

File: src/rpc/mod.rs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        match block_number {
+            types::BlockNumber::Number(n) => n.as_u64(),
+            _ => 0,
+        }
+    }
+    pub fn get_block_number<K: TransactionKind, E: EnvironmentKind>(
+        txn: &MdbxTransaction<'_, K, E>,
+        block_number: ethereum_jsonrpc::types::BlockId,
+    ) -> anyhow::Result<BlockNumber> {
+        let block_number = match block_number {
```
Comment (vorot93) [inline src/rpc/mod.rs:26]: `if let`

File: src/rpc/mod.rs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            tables,
+        },
+        models::{BlockNumber, Message, TransactionAction},
+        stagedsync::stages,
+    };
+    use ethereum_jsonrpc::types;
+    use ethereum_types::U64;
+
+    fn block_number_to_u64(block_number: types::BlockNumber) -> u64 {
+        match block_number {
```
Comment (vorot93) [inline src/rpc/mod.rs:17]: `if let`

**Review by vorot93 — CHANGES_REQUESTED** [review]

File: src/rpc/eth.rs [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+            },
+            sender: call_data.from.unwrap_or_else(Address::zero),
+        };
+        let mut db = InMemoryState::default();
+        let mut state = IntraBlockState::new(&mut db);
+        let mut cache = AnalysisCache::default();
+        let block_spec = MAINNET.collect_block_spec(block_number);
+        let mut tracer = NoopTracer;
+        Ok(U64::from(
+            50_000_000
```
[the remark this answers is not in the document]
Comment (Giulio2002) [reply src/rpc/eth.rs:137]: what about making gaslimit equivalent to header's gas limit?

**Review by vorot93 — APPROVED** [review]

**Pull request merged**